setup.py

```diff
@@ -42,20 +51,9 @@
         exclude=['ci'],
     ),
     extras_require={
-        'test': [
-            'pytest',
-            'pytest-mock',
-        ],
-        'dev': [
-            'pytest',
-            'pytest-mock',
-            'dagger.io',
-            'pre-commit',
-        ],
-        'doc': [
-            'sphinx',
-            'myst-parser',
-        ],
+        'dev': required_dev,
+        'doc': required_doc,
+        'test': required_test,
```

Comment:
Factorize in one function that checks for requirements-*.txt and returns a dict (example [here](https://github.com/great-expectations/great_expectations/blob/develop/setup.py#L45))